### ISSUE TITLE
Link rewrites on javascript links

### ISSUE THREAD

**Issue by samtingleff** [digi-trust/dt-cdn#66]
Link rewriting redirect behavior should only be enabled for http/https href values. Currently the check on this line
https://github.com/digi-trust/dt-cdn/blob/master/src/modules/helpers.js#L212 looks for '#' as a possible javascript link. This is insufficient, for example `javascript:` links.

The behavior should be to only enable redirects for 'http' and 'https' links.

**Comment by samtingleff** [comment]
@goosemanjack can you review please

**Comment by goosemanjack** [comment]
You should avoid String.startsWith as this is not supported in Internet Explorer <= 11 except via polyfill (Edge supports it, but no one uses Edge).

Instead use this, replacing the two lines.

```
var possibleHref = _getElementHref(t) || '';
var posA = possibleHref.indexOf('http://'),
      posB = possibleHref.indexOf('https://'),
      isLink = posA == 0 || posB == 0;
if (isLink) {
   ...
}
```


        if (possibleHref && ((possibleHref.startsWith("http://")) || (possibleHref.startsWith("https://")))) {

**Comment by samtingleff** [comment]
thanks @goosemanjack ! this is released

**Comment by samtingleff** [comment]
The code above to check for http/https links is called on the output of `_getElementHref()`, which can return relative links, which do not necessarily begin with http:// or https://. Suggest modifying `_getElementHref()` to return an absolute link, rather than using `getAbsolutePath()` later.

**Comment by samtingleff** [comment]
@goosemanjack please review again

**Comment by goosemanjack** [comment]
Looks good WRT line changes. Didn't run the whole file to verify as I assume you have already.

**Comment by samtingleff** [comment]
released in 1.5.31
